File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/javascript/JavascriptImportsSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.javascript;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavascriptImportsSettings.class)
public interface JavascriptImportsSettings {

  Optional<Boolean> usePathsRelativeToTheProjectResourceOrSourcesRoots();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/javascript/JavascriptImportsSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.javascript

import com.fasterxml.jackson.annotation.JsonProperty

data class JavascriptImportsSettings(
    @JsonProperty("Use paths relative to the project, resource or sources roots")
    val usePathsRelativeToTheProjectResourceOrSourcesRoots: Boolean? = null
)
